lib/manageiq/api/client/mixins/action_mixin.rb

```diff
@@ -2,6 +2,16 @@ module ActionMixin
   extend ActiveSupport::Concern
 
   private
+  def find_action(name)
+    @action = @actions.find { |action| action.name == name.to_s }
+  end
+
+  def do_action(name)
+    @action ||= find_action(name)
+    client.send @action.method, URI(@action.href) do
+      { action: @action.name }
+    end
```

Comment:
should have an ActionResponse object, so callers can query res.success, res.message, etc.